### PR TITLE
feat: return all notifications even if they have been read

### LINKED ISSUE #229: Feature: return the list of notifications even after they have been read
### Type of feature

🍕 Feature

### Current behavior

https://github.com/open-sauced/insights/assets/79809121/46558a85-ad76-4551-bd36-14fc0cb0a287

### Suggested solution

Return the list of notifications after they have been read by the user for a better experience.

### Additional context

related https://github.com/open-sauced/insights/issues/1444

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by takanome-dev**
## Description

This PR refactors the `UserNotificationService` to return all the notifications even if they have been read.

In `UserNotificationController`, the `OmitType` decorator from `@nestjs/swagger` has been added to exclude the `read_at` property from the response type of the `getUserNotifications` endpoint.

_Generated using [OpenSauced](https://opensauced.ai/)._

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #229 
## Mobile & Desktop Screenshots/Recordings

The `read_at` property has been removed from the response example because it's not returned.
| | |
| ------ | ------ |
| notif-res-body | notif-res-exple |

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by OgDev-01 — APPROVED** [review]
LGTM

File: src/user/user-notifcation.service.ts [hunk 1]
```diff
@@ -29,8 +29,7 @@ export class UserNotificationService {
     queryBuilder
       .innerJoin("users", "users", "user_notifications.user_id=users.id")
       .where("user_id = :userId", { userId })
-      .andWhere("user_notifications.type IN (:...userNotificationTypes)", { userNotificationTypes })
-      .andWhere("user_notifications.read_at IS NULL");
+      .andWhere("user_notifications.type IN (:...userNotificationTypes)", { userNotificationTypes });
```
Comment (brandonroberts) [inline src/user/user-notifcation.service.ts:32]: This is going to cause all the notifications to be marked as read for the user every time this endpoint is fetched, which is not what we want. I would recommend filtering the `notificationIds` to only include notifications that have not been read for marking as read also.

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: src/user/user-notifcation.service.ts (hunk 1 above)
Comment (takanome-dev) [reply src/user/user-notifcation.service.ts:32]: Yeah, I totally missed that, thanks for noticing. In that case, we would want to return the `read_at` property on select queries to properly filter the notifications.

Then, are we going to remove it again before returning the results to the user @brandonroberts?

Comment (brandonroberts) [reply src/user/user-notifcation.service.ts:32]: Why does it need to be removed?

Comment (takanome-dev) [reply src/user/user-notifcation.service.ts:32]: Seeing that the API isn't returning it, I thought for some reason we didn't want to return it.

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**